modules/react/button/lib/SecondaryButton.tsx

```diff
@@ -20,96 +20,104 @@ export interface SecondaryButtonProps extends ButtonProps {
-        [buttonVars.hover.label]: base.blackPepper500,
-        [systemIconStencil.vars.color]: base.blackPepper500,
-      },
-      // Active Styles
-      '&:active, &.active': {
-        [buttonVars.active.background]: base.soap400,
-        [buttonVars.active.border]: base.soap400,
-        [buttonVars.active.label]: base.blackPepper500,
-        [systemIconStencil.vars.color]: base.blackPepper500,
-      },
-      // Disabled Styles
-      '&:disabled, &.disabled': {
-        [buttonVars.disabled.background]: 'transparent',
-        [buttonVars.disabled.border]: base.frenchVanilla100,
-        [buttonVars.disabled.label]: base.frenchVanilla100,
-        [systemIconStencil.vars.color]: base.frenchVanilla100,
+  modifiers: {
+    variant: {
+      inverse: {
+        // Default Styles
+        [baseButtonStencil.vars.background]: 'transparent',
+        [baseButtonStencil.vars.border]: system.color.border.inverse,
+        [baseButtonStencil.vars.label]: system.color.fg.inverse,
+        [systemIconStencil.vars.color]: system.color.fg.inverse,
+        // Focus Styles
+        '&:focus-visible, &.focus': {
+          [baseButtonStencil.vars.background]: system.color.bg.default,
+          [baseButtonStencil.vars.border]: 'transparent',
+          [baseButtonStencil.vars.label]: system.color.fg.strong,
+          [baseButtonStencil.vars.boxShadowInner]: system.color.fg.strong,
+          [baseButtonStencil.vars.boxShadowOuter]: system.color.fg.inverse,
+          [systemIconStencil.vars.color]: system.color.fg.strong,
+        },
+        // Hover Styles
+        '&:hover, &.hover': {
+          [baseButtonStencil.vars.background]: system.color.bg.alt.default,
+          [baseButtonStencil.vars.border]: 'transparent',
+          [baseButtonStencil.vars.label]: system.color.fg.stronger,
+          [systemIconStencil.vars.color]: system.color.fg.stronger,
+        },
+        // Active Styles
+        '&:active, &.active': {
+          [baseButtonStencil.vars.background]: system.color.bg.alt.strong,
+          [baseButtonStencil.vars.border]: 'transparent',
+          [baseButtonStencil.vars.label]: system.color.fg.stronger,
+          [systemIconStencil.vars.color]: system.color.fg.stronger,
+        },
+        // Disabled Styles
+        '&:disabled, &.disabled': {
+          [baseButtonStencil.vars.background]: 'transparent',
+          [baseButtonStencil.vars.border]: system.color.border.inverse,
+          [baseButtonStencil.vars.label]: system.color.fg.inverse,
+          [systemIconStencil.vars.color]: system.color.fg.inverse,
+        },
       },
-    }),
+    },
   },
 });
 
 export const SecondaryButton = createComponent('button')({
   displayName: 'SecondaryButton',
-  Component: ({children, variant, ...elemProps}: SecondaryButtonProps, ref, Element) => {
+  Component: (
+    {children, variant, size, iconPosition, ...elemProps}: SecondaryButtonProps,
+    ref,
+    Element
+  ) => {
     return (
       <Button
```

Comment:
Do you need to use `Button` component here as you extending stencil? All styles from `BaseButton` should be applied and there is no need to pass `size` and `iconPosition` again. This is how extending text works, I'm not sure how it will work on buttons.

Reply:
So for Buttons, we are using `Button` here for a lot of the iconPositions within button (start, end, only).

Reply:
Ahh I see what you mean. 

Let me check and see how this will look. Thanks Raisa!